v2/components/form-with-steps.jsx

```diff
@@ -0,0 +1,149 @@
+import { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useSteps } from '../hooks';
+import { colors, breakpoints } from '../assets/styles';
+import { Button } from './UI';
+import Form from './form';
+
+const FormWithSteps = ({ title, steps, onCompleteAll }) => {
+  const { currentStep, isFinalStep, stepForward, stepBack, steppedBack } = useSteps(steps.length);
+
+  if (title) {
+    useEffect(() => {
+      document.title = `${title} | ${currentStep + 1}. ${steps[currentStep].title}`;
+    }, [currentStep]);
+  }
+
+  /*
+   * Result contains values of all form steps
+   */
+  const [result, setResult] = useState({});
+
+  /*
+   * Add current step values to result & continue
+   */
+  const submitStep = (values) => {
+    /*
+     * Add current step values to multi step form result
+     */
+    setResult({ ...result, ...values });
+
+    /*
+     * Continue one step
+     */
+    stepForward();
+  };
+
+  const handleStepBack = () => {
+    /*
+     * Go back one step
+     */
+    stepBack();
+  };
+
+  /*
+   * Create array containing step titles
+   */
+  const stepTitles = steps.reduce((all, step) => all.concat(step.title), []);
+
+  /*
+   * Current step props
+   */
+  const { stepTitle, fields } = steps[currentStep];
+
+  const stepsLeft = steps.length - currentStep - 1;
+
+  return (
+    <Form onSubmit={submitStep} title={stepTitle} fields={fields} isPreviousStep={steppedBack}>
+      <div className="stepper">
+        {currentStep > 0 && (
+          <Button type="button" onClick={handleStepBack}>
+            Stap terug
+          </Button>
+        )}
+        <div className="step-titles">
+          {stepTitles.map((title, i) => (
+            <p key={i} className={i === currentStep && 'active'}>
+              {title}
+            </p>
+          ))}
```

Comment:
`title` is already declared in the upper scope
```suggestion
          {stepTitles.map((stepTitle, i) => (
            <p key={i} className={i === currentStep && 'active'}>
              {stepTitle}
            </p>
          ))}
```